src/jace/stages.py

```diff
@@ -50,203 +51,225 @@
-    Furthermore, a `JaCeWrapped` object is composable with all Jax transformations.
+    used. While it supports just-in-time lowering and compilation, by just
+    calling it, these steps can also be performed explicitly.
+    The lowering, performed by this stage is cached, thus if a `JaCeWrapped`
+    object is later lowered with the same arguments the result might be taken
+    from the cache.
+
+    Furthermore, a `JaCeWrapped` object is composable with all Jax transformations,
+    all other stages are not.
 
     Args:
         fun: The function that is wrapped.
         primitive_translators: Primitive translators that that should be used.
         jit_options: Options to influence the jit process.
 
     Todo:
-        - Support pytrees.
-        - Support keyword arguments and default values of the wrapped function.
+        - Support default values of the wrapped function.
         - Support static arguments.
 
     Note:
         The tracing of function will always happen with enabled `x64` mode,
-        which is implicitly and temporary activated while tracing.
+        which is implicitly and temporary activated during tracing.
     """
 
-    _fun: Callable
+    _fun: Callable[_P, _RetrunType]
     _primitive_translators: dict[str, translator.PrimitiveTranslator]
-    _jit_options: dict[str, Any]
+    _jit_options: api.JitOptions
 
     def __init__(
         self,
-        fun: Callable,
+        fun: Callable[_P, _RetrunType],
         primitive_translators: Mapping[str, translator.PrimitiveTranslator],
-        jit_options: Mapping[str, Any],
+        jit_options: api.JitOptions,
     ) -> None:
+        assert all(
+            param.default is param.empty for param in inspect.signature(fun).parameters.values()
+        )
         super().__init__()
-        # We have to shallow copy both the translator and the jit options.
-        #  This prevents that any modifications affect `self`.
-        #  Shallow is enough since the translators themselves are immutable.
         self._primitive_translators = {**primitive_translators}
-        # TODO(phimuell): Do we need to deepcopy the options?
         self._jit_options = {**jit_options}
         self._fun = fun
 
-    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+    def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _RetrunType:
         """
         Executes the wrapped function, lowering and compiling as needed in one step.
 
-        The arguments passed to this function are the same as the wrapped function uses.
+        This function will lower and compile in one go. The function accepts the same
+        arguments as the original computation and the return value is unflattened.
+
+        Note:
+            This function is also aware if a Jax tracing is going on. In this
+            case, it will forward the computation.
+            Currently, this function ignores the value of `jax.disable_jit()`.
```

Comment:
If `jit` is disabled JAX will not compile or reuse the compiled function instead it would run the python computation.
I think this is essentially a debug statement to check if the compiler screwed up.

The question of why it is ignored is because currently every JAX and for that mater DaCe configuration variable is ignored. 
You are right we should in the long term adhere to it.
My "problem," or rather the thing I have not sorted out yet, is that JaCe would then be configurated by two configurations once from DaCe and once from JAX.
Thus I think the best choice is to make things not configurable, at least for now and add it later.
However, I am open to add it in this PR, what are you thinking?